frontend/src/App.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
+import BusList from './components/BusList';
+import './App.css';
+
+function App() {
+  const [buses, setBuses] = useState([]);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  useEffect(() => {
+    console.log('Fetching bus data...');
+    fetch('http://localhost:3000/api/buses')
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then(data => {
+        setBuses(data);
+      })
+      .catch(error => {
+        console.error('Error fetching data:', error);
+      });
+  }, []);
+  
+  // Filter buses based on the search term
+  const filteredBuses = buses.filter(bus =>
+    bus.busNumber.toLowerCase().includes(searchTerm.toLowerCase()) ||
```

Comment:
```suggestion
    bus.busNumber.toLowerCase().includes(searchTerm.toLowerCase()) ||
    bus.busNumber?.toLowerCase().includes(searchTerm.toLowerCase())
```
This will prevent any errors if busNumber is undefined or null. Similarly, check for bus.from, bus.to, and bus.routes properties before using them.